builtins.oomph

```diff
@@ -38,70 +53,64 @@ func __Str_center_pad(Str self, Int len, Str pad_char) -> Str:
     return self.left_pad(average, pad_char).right_pad(len, pad_char)
 
 func __Str_count(Str self, Str substring) -> Int:
-
     if substring == "":
-        # Same as what Python does
-        return self.length() + 1
+        # Consistent with .split("")
+        return self.length() - 1
 
-    # TODO: this loop isn't elegant, ideally condition could be 'i := self.find_first(...)'
-    for let count = 0; ; count = count + 1:
-        let i = self.find_first(substring)
-        if i == null:
+    let count = 0
+    while true:
+        let before_substring = __string_from_start_to_substring(self, substring)
+        if before_substring == self:
             return count
-        self = self.slice(i.get() + substring.length(), self.length())
-
-func starts_with_space(Str s) -> Bool:
-    # TODO: all ascii whitespace
-    return s.starts_with(" ") or s.starts_with("\n")
+        count = count + 1
+        self = self.remove_prefix(before_substring).get().remove_prefix(substring).get()
 
 func __Str_left_trim(Str self) -> Str:
-    while starts_with_space(self):
-        self = self.slice(1, self.length())
+    # TODO: all ascii whitespace?
+    while self.starts_with(" ") or self.starts_with("\n"):
+        if self.starts_with(" "):
+            self = self.remove_prefix(" ").get()
+        if self.starts_with("\n"):
+            self = self.remove_prefix("\n").get()
     return self
 
 func __Str_right_trim(Str self) -> Str:
-    while self.ends_with(" "):   # FIXME: something like starts_with_space
-        self = self.slice(0, self.length() - 1)
+    # TODO: left_trim copy/pasta
+    while self.ends_with(" ") or self.ends_with("\n"):
+        if self.ends_with(" "):
+            self = self.remove_suffix(" ").get()
+        if self.ends_with("\n"):
+            self = self.remove_suffix("\n").get()
     return self
 
 func __Str_trim(Str self) -> Str:
     return self.left_trim().right_trim()
 
-func __Str_replace(Str self, Str old, Str neuw) -> Str:
-    if old == "":
-        # TODO: error
-        return self
-
-    let result = ""
-    let remaining = self
-
-    while true:
-        let index = remaining.find_first(old)
-        if index == null:
-            return result + remaining
-        result = result + remaining.slice(0, index.get()) + neuw
-        remaining = remaining.slice(index.get() + old.length(), remaining.length())
+func __Str_replace(Str self, Str old, Str neww) -> Str:
+    return self.split(old).join(neww)
 
 func __Str_split(Str self, Str sep) -> List[Str]:
-    let result = new List[Str]()
+    let result = []
     if sep == "":
-        result.push(self)
+        # get list of unicode characters
+        while self != "":
+            result.push(__string_get_first_char(self))
+            self = self.remove_prefix(result.last()).get()
```

Comment:
Looks like every `.remove_prefix()` call ends with `.get()`. Maybe just have it not return an optional?